frame/dapps-staking/src/testing_utils.rs

```diff
@@ -394,43 +394,68 @@ pub(crate) fn assert_withdraw_unbonded(staker: AccountId) {
 /// Used to perform claim for stakers with success assertion
 pub(crate) fn assert_claim_staker(claimer: AccountId, contract_id: &MockSmartContract<AccountId>) {
     let (claim_era, _) = DappsStaking::staker_info(&claimer, contract_id).claim();
-    let init_state = MemorySnapshot::all(claim_era, contract_id, claimer);
+    let current_era = DappsStaking::current_era();
+
+    //clean up possible leftover events
+    System::reset_events();
+
+    let init_state_claim_era = MemorySnapshot::all(claim_era, contract_id, claimer);
+    let init_state_current_era = MemorySnapshot::all(current_era, contract_id, claimer);
 
     // Calculate contract portion of the reward
-    let (_, stakers_joint_reward) =
-        DappsStaking::dev_stakers_split(&init_state.contract_info, &init_state.era_info);
+    let (_, stakers_joint_reward) = DappsStaking::dev_stakers_split(
+        &init_state_claim_era.contract_info,
+        &init_state_claim_era.era_info,
+    );
 
-    let (claim_era, staked) = init_state.staker_info.clone().claim();
+    let (claim_era, staked) = init_state_claim_era.staker_info.clone().claim();
     assert!(claim_era > 0); // Sanity check - if this fails, method is being used incorrectly
 
     // Cannot claim rewards post unregister era, this indicates a bug!
-    if let DAppState::Unregistered(unregistered_era) = init_state.dapp_info.state {
+    if let DAppState::Unregistered(unregistered_era) = init_state_claim_era.dapp_info.state {
```

Comment:
well, yeah, it took me a while to figure out snapshots capture a specific era :)
I created two separate snapshots, one for `claim_era`, one for `current_era`, do you mind double checking the logic for the assertions now, in case I messed something up?

Reply:
I did during review :)

It looks good to me!